test/basics_datamunger_test.js

```diff
@@ -19,69 +19,285 @@ var chai = require('chai');
 var assert = chai.assert;
 var expect = chai.expect;
 
+var _ = require('lodash');
+var d3 = require('d3');
+
 var dm = require('../plugins/blip/basics/logic/datamunger');
 
+var types = require('../dev/testpage/types');
+
 describe('basics datamunger', function() {
   it('should be an object', function() {
     assert.isObject(dm);
   });
 
   describe('bgDistribution', function() {
+    var bgClasses = {
+      'very-low': {boundary: 10},
+      low: {boundary: 20},
+      target: {boundary: 30},
+      high: {boundary: 40},
+      'very-high': {boundary: 50}
+    };
+    var zeroes = {
+      veryhigh: 0,
+      high: 0,
+      target: 0,
+      low: 0,
+      verylow: 0
+    };
     it('should be a function', function() {
       assert.isFunction(dm.bgDistribution);
     });
 
-    it('should always calculate a BG distribution for smbg data');
-
-    it('should yield cgmStatus `noCGM` if no cbg data');
+    it('should always calculate a BG distribution for smbg data, should calculate a BG distribution for cbg data if averages >= 144 readings per day, and should yield cgmStatus `calculatedCGM` when have calculated a BG distribution for cbg data', function() {
+      var now = new Date();
+      var smbg = [
+        new types.SMBG({value: 25})
+      ];
+      var cbg = [];
+      for (var i = 0; i < 144; ++i) {
+        cbg.push(new types.CBG({
+          deviceTime: new Date(now.valueOf() + i*2000).toISOString().slice(0,-5),
+          value: 50
+        }));
+      }
+      expect(dm.bgDistribution({
+        data: {smbg: {data: smbg}, cbg: {data: cbg}},
+        dateRange: [d3.time.day.utc.floor(now), d3.time.day.utc.ceil(now)]
+      }, bgClasses)).to.deep.equal({
+        cbg: _.defaults({veryhigh: 1}, zeroes),
```

Comment:
Yup, in line with the idea of the components being dumb as rocks (== do no calculation themselves).